### PR TITLE
Updates for Apple Silicon

### PR DESCRIPTION
**Pull request by stefansundin**
This is based on top of #14. So when that PR is merged I can rebase this branch for a smaller diff.

- Search a list of paths to find the Homebrew executable, since Homebrew installs to a different path by default on Apple Silicon. The `UserDefaults` code can handle both a string and an array now, for backwards-compatibility.
- Show `Homebrew now found` instead of `No services available` when the executable cannot be found.
- Add a `Homebrew error` menu item if `brew services list` returns a non-zero exit code.
- Allow a service in `error` status to be stopped.

Screenshots of new menu items:

Screen Shot 2021-04-18 at 09 38 59

Screen Shot 2021-04-18 at 09 39 40

After building the binary, you can verify that it is built for both architectures by running:
```
% lipo -archs DerivedData/BrewServicesMenubar/Build/Products/Release/BrewServicesMenubar.app/Contents/MacOS/BrewServicesMenubar
x86_64 arm64
```

### CODE REVIEW

**Comment by andrewn** [comment]
@stefansundin Would you mind rebasing this now that #14 is merged please? I have Xcode setup now so will try to review it ASAP.

**Comment by stefansundin** [comment]
@andrewn Rebased 👍

**Comment by stefansundin** [comment]
By the way, I added `xcuserdata` to `.gitignore`, but I realized now that you have a directory there. Not sure if that directory should be checked in or not, but my goal was to prevent my own `xcuserdata/stefan.xcuserdatad` from being committed. Feel free to remove that line from `.gitignore`.

**Comment by andrewn** [comment]
> By the way, I added `xcuserdata` to `.gitignore`, but I realized now that you have a directory there. Not sure if that directory should be checked in or not, but my goal was to prevent my own `xcuserdata/stefan.xcuserdatad` from being committed. Feel free to remove that line from `.gitignore`.

I'm not very knowledgeable about development with XCode so including `xcuserdata` is due to that!

File: BrewServicesMenubar/AppDelegate.swift [hunk 1]
```diff
@@ -26,9 +30,17 @@ class AppDelegate: NSObject, NSApplicationDelegate {
     var services: [Service]?
 
     func applicationDidFinishLaunching(_ aNotification: Notification) {
-        UserDefaults.standard.register(defaults: [
-            brewExecutableKey: "/usr/local/bin/brew"
-        ])
+        // Homebrew can now be located in two different locations depending on how it was installed and what architecture the computer is running
+        // Define the most likely path first based on the architecture
+        #if arch(arm64)
```
Comment (andrewn) [inline BrewServicesMenubar/AppDelegate.swift:36]: Is the architecture check necessary? I'm just wondering what the implications are if we have the same order for both platforms? e.g. `"/opt/Homebrew/bin/brew", "/usr/local/bin/brew"`?

**Comment by andrewn** [comment]
This looks and works great, thanks so much for this!

I'm curious about swapping the list of default executable paths per-architecture, but I've merge anyway.

File: BrewServicesMenubar/AppDelegate.swift (hunk 1 above)
Comment (stefansundin) [reply BrewServicesMenubar/AppDelegate.swift:36]: It's just a tiny optimization. We'd just do one unnecessary file existence check every time we try to invoke `brew`. The app should work fine without it.

**Pull request merged**